kodilla-good-patterns/src/main/java/com/kodilla/good/patterns/orders/ordering/OrderProcessor.java

```diff
@@ -0,0 +1,35 @@
+package com.kodilla.good.patterns.orders.ordering;
+
+import com.kodilla.good.patterns.orders.ordering.services.*;
+
+public class OrderProcessor {
+    private final InformationService infoService;
+    private final OrderService orderService;
+    private final OrderRepository orderRepo;
+
+    public OrderProcessor(final InformationService infoService,
+                          final OrderService orderService,
+                          final OrderRepository orderRepo) {
+        if (infoService == null || orderService == null || orderRepo == null) {
+            throw new IllegalArgumentException();
+        }
+
+        this.infoService = infoService;
+        this.orderService = orderService;
+        this.orderRepo = orderRepo;
+    }
+
+    public OrderProcessResultDto process(final ProductOrder order) {
+        if (order == null) {
+            throw new IllegalArgumentException();
+        }
+
+        PlacedProductOrder ppo = orderService.order(order);
+        if (ppo == null) {
```

Comment:
in this case we should return null or throw Exception

Reply:
Yes,  I realise the error is too serious to be logged only.